CONTRIBUTING.md

```diff
@@ -190,15 +190,25 @@ of their patch is not influenced by their setup or implicit knowledge of the sys
 
 ## Code Style
 
-We use automated formatting with Fourmolu to enforce a unified style across the code bases. It is checked in the CI process.
-After installing Fourmolu 0.12, there are some makefile targets to help formatting
-the code base.
-
+We use automated formatting, checked in the CI process, to enforce a unified
+style across the code bases and include these makefile targets to help with
+formatting:
 
 * `make style` - Format the `Cabal`, `Cabal-syntax` and `cabal-install` directories.
 * `make style-modified` - Format files modified in the current tree.
 * `make style-commit COMMIT=<ref>` - Format files modified between HEAD and the given reference.
 
+> [!NOTE]
+> We use `fourmolu-0.12.0.0` for formatting. If installing it with `cabal
+> install`, please make sure to use a version of `GHC >= 9.2.1 && < 9.8`. That
+> it requires `GHC2021` sets the lower bound and its reliance on
+> `ghc-lib-parser` sets the upper bound on GHC versions. The command for
+> installing it this way is:
+>
+> ```
+> $ cabal install fourmolu-0.12.0.0 --overwrite-policy=always --ignore-project
+> ```
+
```

Comment:
```suggestion

(This assumes you have a ghc version in the 9.6 series installed via ghcup.)
```

Reply:
ghcup is not available everywhere, e.g. NixOS, so I'd not overfit for it.

Reply:
I provided it as an example, and it does provide that symlink. I don't think we can provide examples for every possible distribution, but I also don't think we should provide none, so I provided the most common.

Reply:
In fact, nix would require starting with an appropriate `nix-shell` invocation AIUI, so we'd have trouble providing that example anyway.

Reply:
Not necessarily: `ghc` can be installed in `systemPackaes` on a NixOS system.

Reply:
But it may be the wrong version, especially since you need an older one for fourmolu 0.12.

I am getting the strong impression that you think hanging new contributors out to dry is better than providing _any_ example. _Really?_

Reply:
Also, the specific reason for `--with-compiler` is that the original says nothing whatsoever about making sure an appropriate compiler version is in scope for the `cabal install`.

Reply:
> the original says nothing whatsoever about making sure an appropriate compiler version is in scope

Yes, I went for an incremental improvement and mentioned that the GHC version will need to be restricted. I didn't go so far as to say how to go about that but am happy to. I use ghcup for that (and sometimes nix).

Reply:
I don't suggest removing the example. I only suggest providing essential details and leaving off inessential or potentially problematic ones. For example, the way people obtain GHC is inessential (also, hardcoding a particular distribution channel is not very future-proof). The `with-compiler` is completely optional, as long as the text clearly states a particular version of GHC that should be in scope.

I don't want to drag this thread for too long, so if you feel strongly about `with-compiler` and mentioning ghcup, please go ahead.

Reply:
Looking at the package bounds on hackage, `fourmolu-0.12.0.0` requires `ghc-lib-parser (>=9.6 && <9.7)` but that doesn't restrict GHC to just the `9.6` series. The constraints on `base` for [ghc-lib-parser-9.6.6.20240701](https://hackage.haskell.org/package/ghc-lib-parser-9.6.6.20240701) are `base (>=4.16.1 && <4.19)`, so from [snoyman/base](https://www.snoyman.com/base/) that is `GHC >= 9.2.2 && GHC < 9.8`.